src/main/java/org/kohsuke/github/GHHook.java

```diff
@@ -60,4 +50,8 @@ public void delete() throws IOException {
     public URL getHtmlUrl() {
         return null;
     }
+
+    abstract GitHub root();
```

Comment:
* Implement methods with a default behavior instead.
* Oracle Java codestyle requires getRoot() and getPath()
* Javadocs are required as well. Also add <code>@since TODO</code> to document the API changes properly